### PR TITLE
Cors

### PR DESCRIPTION
**Pull request by eulogio-gutierrez**
Decorator to add CORS headers to a decorated function response dictionary

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -26,15 +26,16 @@ The easiest way to use these AWS Lambda Decorators is to install them through Pi
...
+* [__log__](https://github.com/gridsmartercities/aws-lambda-decorators/blob/master/aws_lambda_decorators/decorators.py#L122-L133): a decorator to log the parameters passed to the lambda function and/or the response of the lambda function.
+* [__handle_exceptions__](https://github.com/gridsmartercities/aws-lambda-decorators/blob/master/aws_lambda_decorators/decorators.py#L94-L119): a decorator to handle any type of declared exception generated by the lambda function. 
+* [__response_body_as_json__](https://github.com/gridsmartercities/aws-lambda-decorators/blob/master/aws_lambda_decorators/decorators.py#L164-L183): a decorator to transform a response dictionary body to a json string.
+* [__handle_all_exceptions__](https://github.com/gridsmartercities/aws-lambda-decorators/blob/master/aws_lambda_decorators/decorators.py#L212-L232): a decorator to handle all exceptions thrown by the lambda function.
+* [__cors__](https://github.com/gridsmartercities/aws-lambda-decorators/blob/master/aws_lambda_decorators/decorators.py#L235-L288): a decorator to handle all exceptions thrown by the lambda function.
```
Comment (danPudwellUK) [inline README.md]: description needs amending, copied from previous

File: README.md [hunk 2]
```diff
@@ -321,6 +322,21 @@ def response_body_as_json_example():
...
+
+* Returns a 500 error if one or more of the CORS headers have an invalid type
+
+Example: [code](https://github.com/gridsmartercities/aws-lambda-decorators/blob/master/examples/examples.py#L124-L126) | [test](https://github.com/gridsmartercities/aws-lambda-decorators/blob/master/examples/test_examples.py#L217-L227)
+```python
+@cors(allow_origin='*', allow_methods='POST', allow_headers='Content-Type', max_age=86400)
+def cors_example():
+    return {'statusCode': 200}
+    
+cors_example()  # returns {'statusCode': 200, 'headers': {'access-control-allow-origin': '*', 'access-control-allow-methods': 'POST', 'access-control-allow-headers': 'Content-Type', 'access-control-max-age': 86400}}
```
Comment (danPudwellUK) [inline README.md]: are these case-sensitive?

File: aws_lambda_decorators/decorators.py [hunk 3]
```diff
@@ -226,3 +230,59 @@ def wrapper(*args, **kwargs):
...
+
+            response = func(*args, **kwargs)
+
+            if isinstance(response, dict):
+                headers_key = find_key_case_insensitive('headers', response)
+
+                resp_headers = response[headers_key] if headers_key in response else {}
+
+                try:
+                    resp_headers = update_header(resp_headers, 'access-control-allow-origin', allow_origin, str)
```
Comment (danPudwellUK) [inline aws_lambda_decorators/decorators.py]: case sensitive?

File: README.md (hunk 1 above)
Comment (eulogio-gutierrez) [reply README.md]: fixed!

File: README.md (hunk 2 above)
Comment (eulogio-gutierrez) [reply README.md]: they are all case-insensitive

Comment (eulogio-gutierrez) [reply README.md]: https://stackoverflow.com/questions/5258977/are-http-headers-case-sensitive

File: aws_lambda_decorators/decorators.py (hunk 3 above)
Comment (eulogio-gutierrez) [reply aws_lambda_decorators/decorators.py]: https://stackoverflow.com/questions/5258977/are-http-headers-case-sensitive

**Pull request merged**